contracts/modules/ProtocolSettings.sol

```diff
@@ -393,6 +395,30 @@ contract ProtocolSettings is
         emit SetFeesController(msg.sender, oldController, newController);
     }
 
+    /**
+     * @notice Set the admin address of sovryn protocol.
+     *
+     * only owner can perform this action.
+     *
+     * @param newAdmin The new address of the admin.
+     * */
+    function setAdmin(address newAdmin) external onlyOwner {
+        address oldAdmin = admin;
+        admin = newAdmin;
+
+        emit SetAdmin(msg.sender, oldAdmin, newAdmin);
```

Comment:
emitting the event first eliminates need for a local variable oldAdmin and readin/writing

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/511/commits/2c7cab319bc4d15bc3a796a3805f41b5786313e5)